components/org.wso2.carbon.identity.conditional.auth.functions.choreo/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/choreo/ClientManager.java

```diff
@@ -0,0 +1,175 @@
+
+    }
+
+    /**
+     * Get HTTPClient properly configured with tenant configurations.
+     *
+     * @param tenantDomain tenant domain of the service provider.
+     * @return HttpClient
+     */
+    public CloseableHttpAsyncClient getClient(String tenantDomain) throws FrameworkException, IOException {
+
+        int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
+        CloseableHttpAsyncClient client = clientMap.get(tenantId);
+
+        if (client == null) {
+
+            RequestConfig config = createRequestConfig();
+
+            HttpAsyncClientBuilder httpClientBuilder = HttpAsyncClients.custom().setDefaultRequestConfig(config);
+
+            addSslContext(httpClientBuilder, tenantDomain);
+            httpClientBuilder.setConnectionManager(poolingHttpClientConnectionManager);
+
+            client = httpClientBuilder.build();
+            client.start();
+            clientMap.put(tenantId, client);
+        }
+
+        return client;
+    }
+
+    private RequestConfig createRequestConfig() {
+
+        return RequestConfig.custom()
+                .setConnectTimeout(HTTP_CONNECTION_TIMEOUT)
+                .setConnectionRequestTimeout(HTTP_CONNECTION_REQUEST_TIMEOUT)
+                .setSocketTimeout(HTTP_READ_TIMEOUT)
+                .build();
+    }
+
+    private PoolingNHttpClientConnectionManager createPoolingConnectionManager() throws FrameworkException {
+
+        String maxConnectionsString = IdentityUtil.getProperty(Constants.CONNECTION_POOL_MAX_CONNECTIONS);
+        String maxConnectionsPerRouteString = IdentityUtil.getProperty(Constants
+                .CONNECTION_POOL_MAX_CONNECTIONS_PER_ROUTE);
+        int maxConnections = DEFAULT_MAX_CONNECTIONS;
+        int maxConnectionsPerRoute = DEFAULT_MAX_CONNECTIONS;
+        try {
+            maxConnections = Integer.parseInt(maxConnectionsString);
+        } catch (NumberFormatException e) {
+            // Default value is used.
+            LOG.error("Error while converting MaxConnection " + maxConnections + " to integer. So proceed with " +
+                    "default value ", e);
+        }
+        try {
+            maxConnectionsPerRoute = Integer.parseInt(maxConnectionsPerRouteString);
+        } catch (NumberFormatException e) {
+            // Default value is used.
+            LOG.error("Error while converting MaxConnectionsPerRoute " + maxConnectionsPerRoute + " to integer. " +
+                    "So proceed with default value ", e);
+        }
+
+        ConnectingIOReactor ioReactor;
+        try {
+            ioReactor = new DefaultConnectingIOReactor();
+        } catch (IOReactorException e) {
+            throw new FrameworkException("Error while creating ConnectingIOReactor", e);
+        }
+        PoolingNHttpClientConnectionManager poolingHttpClientConnectionManager = new
+                PoolingNHttpClientConnectionManager(ioReactor);
+        // Increase max total connection to 20
```

Comment:
Let's format all the in-line comments by adding full-stops. Check all places.

Reply:
Addressed with commit [97b8dc8](https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/85/commits/97b8dc822153f28e50e8f878d46d78b6d6dfd1c0)